cli/tests/test_concurrent_operations.rs

```diff
@@ -227,6 +227,115 @@ fn test_concurrent_snapshot_wc_reloadable() {
     ");
 }
 
+#[test]
+fn test_git_head_race_condition() {
+    // Test for race condition where concurrent jj processes create divergent
+    // operations when importing/exporting Git HEAD. This test spawns two
+    // processes in parallel: one running `jj debug snapshot` repeatedly
+    // (which imports Git HEAD) and another running `jj next/prev` repeatedly
+    // (which exports Git HEAD). Without the fix, this would create divergent
+    // operations.
+
+    let test_env = TestEnvironment::default();
+    test_env
+        .run_jj_in(".", ["git", "init", "--colocate", "repo"])
+        .success();
+    let work_dir = test_env.work_dir("repo");
+
+    // Create a large initial working copy to make snapshotting slower
+    for j in 0..200 {
+        let filename = format!("file_{j}");
+        let content = format!("content for file {j}\n").repeat(100);
+        work_dir.write_file(&filename, &content);
+    }
+    work_dir.run_jj(["commit", "-m", "initial"]).success();
+
+    // Create additional commits to iterate through with jj next/prev
+    for i in 0..10 {
+        for j in 0..50 {
+            let filename = format!("file_{i}_{j}");
+            let content = format!("content for commit {i} file {j}\n").repeat(100);
+            work_dir.write_file(&filename, &content);
+        }
+        work_dir
+            .run_jj(["commit", "-m", &format!("commit {i}")])
+            .success();
+    }
+
+    // Extract environment from TestEnvironment to use in spawned processes
+    let base_cmd = test_env.new_jj_cmd();
+    let jj_bin = base_cmd.get_program().to_owned();
+    let base_env: Vec<_> = base_cmd
+        .get_envs()
+        .map(|(k, v)| (k.to_owned(), v.map(|v| v.to_owned())))
+        .collect();
+
+    let repo_path = work_dir.root().to_path_buf();
+    let duration = std::time::Duration::from_secs(5);
+    let start = std::time::Instant::now();
+
+    let snapshot_handle = {
+        let jj_bin = jj_bin.clone();
+        let repo_path = repo_path.clone();
+        let base_env = base_env.clone();
+        std::thread::spawn(move || {
```

Comment:
nit: I think you can use `std::thread::scope()` and not have to clone the variables and `move` the closure. You would use a single scope for both threads.